force-app/main/default/classes/ESA_cls_caseTriggerHandler.cls

```diff
@@ -289,4 +293,39 @@ public with sharing class ESA_cls_caseTriggerHandler {
             System.debug('Exception caught: ' + err.getMessage());
         }
     }
+    
+/**
+* @author: Deepak
+* @date: 20 Jun 2024
+* @description: The purpose of this method is to Populate EDRD Reference number from CareProgramEnrollee.
+* @Modification Log: [Date] - [Change Reference] - [Changed By] - [Description]
+*/
+    public static void populateEDRDRefNumber(List<Case> newCaseList, Map<Id, Case> newCaseMap, Map<Id, Case> oldCaseMap){
+        Map<Id, Case> patientIdVsCase = new Map<Id, Case>();
+        Map<Id, CareProgramEnrollee> patientIdVsCarePrgEnrolle = new Map<Id, CareProgramEnrollee>();
+        
+        for(Case caseObj: newCaseList){
+            Boolean isAccountIdChanged = trigger.isUpdate && newCaseMap.get(caseObj.Id).accountId != oldCaseMap.get(caseObj.Id).accountId;
+            if(!String.isEmpty(caseObj.AccountId) && caseObj.RecordTypeId == recordTypeCaseEDRD && 
+               (trigger.isInsert || isAccountIdChanged)){
+                   patientIdVsCase.put(caseObj.AccountId, caseObj);
+               }
+        }
+        if(!patientIdVsCase.isEmpty()){
+            List<Account> accList = [SELECT Id, Name, (SELECT Id, Name, CareProgramId FROM AccountCareProgramEnrollees 
```

Comment:
add in selector class

Reply:
@yashrohan-cgi , we do not have a frame setup for selector class.